sql/sql_table.cc

```diff
@@ -2900,12 +2900,25 @@ int mysql_rm_table_no_locks(THD *thd, TABLE_LIST *tables, bool if_exists,
         built_query.append(" /* generated by server */");
         int error_code = (non_tmp_error ?
           (foreign_key_error ? ER_ROW_IS_REFERENCED : ER_BAD_TABLE_ERROR) : 0);
-        error |= thd->binlog_query(THD::STMT_QUERY_TYPE,
+
+        if(thd->variables.binlog_ddl_query_log_events)
```

Comment:
Few questions:
1. Why was it necessary to implement a query builder for only DROP TABLE instead of logging the query as-is?
2. Let's say we try to execute `DROP TABLE t1,t2` where t1 is a normal table and  t2 is a temporary table, Since t2 is a temp table, it would not have been replicated to slave in RBR. In such cases, if we log DROP TABLE t1,t2 as is, then wouldn't it cause replication failure on slave with an error that it could not find table t2.
3. What happens in such cases when we enable the variable and execute `DROP TEMPORARY normal_table, temp_table;` it in a cluster environment like GR/PXC.?

Reply:
@venkatesh-prasad-v I honestly don't feel the necessity of an option. Extra comments in binlog wouldn't break any format changes or introduce disk format incompatibilities. Can you please confirm this?

Reply:
I am open for removing option ,if everyone is fine with this.